ddl/ddl_worker.go

```diff
@@ -833,6 +835,12 @@ func updateSchemaVersion(t *meta.Meta, job *model.Job) (int64, error) {
 			return 0, errors.Trace(err)
 		}
 		diff.TableID = job.TableID
+	case model.ActionExchangeTablePartition:
+		err = job.DecodeArgs(&diff.TableID, &diff.PtSchemaID, &diff.PtTableID)
+		if err != nil {
+			return 0, errors.Trace(err)
+		}
+		diff.OldTableID = job.TableID
```

Comment:
I think we also need to update `OldSchemaID` when the `pt` and `nt` isn't in the same database.

Reply:
While the job sets nt.SchemaID as job.SchemID and nt.TableID as job.TableID, For nt, it doesnot have `OldSchemaID`. I have viewd the code of `ApplyDiff`. I thought it only reload the updated meta info of table or schema and return affected tableID by using the `SchemaDiff`. for nt, It should drop the old meta id and create new, while both under the same schema, And for pt, it should also update to due to partition definition changed. In this situation we need to record follwers in `SchemaDiff`:
- `DefID`: the target id of partition definition which is the new table id of nt
- `job.SchemaID`: it's `nt.SchemaID`
- `job.TableID`: It's `nt.TableID`
- `oldTableID`: the old table id of `nt` which we should drop in `ApplyDiff`
- `ptSchemID` and `ptTableID`:  we shoule reload `pt` dut to changed by exchanging partition which that we shoul exchange in `parser` model.

Even if we use pt as the job.SchemaID and job.TableID, it seems that we should do same

Reply:
I think we needn't add `ptSchemID` and `ptTableID`. "newPartitionID" of "pt" is "oldTableID" of "nt" and "newTableID" of "nt" is "oldPartitionID" of "pt". is it sufficient to record only pt.PartitionID and nt.ID? Schema is similar. `SchemaDiff` fields as follows:
`diff. SchemaID `:  it's `nt.SchemaID`
`diff.TableID`: It's `nt.TableID`
`diff. OldSchemaID `:  it's `pt.SchemaID`
`diff.OldTableID`: It's `pt.TableID`
I think we need to drop the table of `nt`(`diff.SchemaID.TableID`) and `pt`(`diff.OldSchemaID.OldTableID`), and create the new `nt`(`diff.SchemaID.OldTableID`) and new `pt`(`diff.OldSchemaID.TableID`) in `ApplyDiff`. I don’t know if it’s clear

> Even if we use pt as the job.SchemaID and job.TableID, it seems that we should do same
Agree it.

Reply:
> create the new nt(diff.SchemaID.OldTableID)

> -  diff.OldTableID: It's pt.TableID

It seems that the `diff.SchemaID.OldTableID` is the id of `pt`, not the origin partition one

Reply:
![WeChatWorkScreenshot_2f7be5cd-6cd2-4ce6-9e4b-79aed86039ce](https://user-images.githubusercontent.com/4242506/83596585-93690700-a597-11ea-9659-1125369f6638.png)
As can be seen from the above figure: `diff.OldSchemaID.TableID`(`pt.SchemaID`.`nt.TableID`) is new `pt`.
![WeChatWorkScreenshot_f1340fe5-25a0-4499-bcf0-9da93979b9d5](https://user-images.githubusercontent.com/4242506/83596605-9e239c00-a597-11ea-809b-7e7c0a784d79.png)
As can be seen from the above figure: `diff.SchemaID.OldTableID`(`nt.SchemaID`.`pt.TableID`) is new `pt`.

Reply:
`diff(SchemaDiff)` inits from `job.Args`. If you means load new `nt` from meta, it still to know the `new tableID` from `nt`. We need to store `nt.tableID(old)`, `nt.SchemaID`, `partID(new nt.tabelID)`, `pt.TableID`, `pt.SchemaID` in `diff`. I still donot know what you means to omit the  `partID(new nt.tableID)`.

Reply:
We don't need `new partID`, because it is put in `tableInfo`. And we can use `pt.TableID` to get `tableInfo` in `ApplyDiff`.

Reply:
You means compare the `pt.OldTableInfo` and `pt.NewTableInfo` to get the `new nt id`?

Reply:
Yes, maybe this is a little complicated. I had a discussion with @cr
. Is it convenient for you to talk in slack?